subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/PluginUnderTestMetadataIntegrationTest.groovy

```diff
@@ -119,4 +119,28 @@ class PluginUnderTestMetadataIntegrationTest extends AbstractIntegrationSpec {
         def hash3 = GUtil.loadProperties(metadataFile).getProperty(IMPLEMENTATION_CLASSPATH_HASH_PROP_KEY)
         hash3 == hash2
     }
+
+    def "not up-to-date if pluginClasspath change"() {
```

Comment:
need some coverage that when the _content_ of the classpath changes, but the classpath does not change, then the task is up-to-date.